ml/src/svm.cpp

```diff
@@ -2351,14 +2351,8 @@ svm_group_classes(const svm_problem* prob,
     if (j == nr_class) {
       if (nr_class == max_nr_class) {
         max_nr_class *= 2;
-        int* label_temp = static_cast<int*>(realloc(label, max_nr_class * sizeof(int)));
-        if (label_temp) {
-          label = label_temp;
-        }
-        int* count_temp = static_cast<int*>(realloc(count, max_nr_class * sizeof(int)));
-        if (count_temp) {
-          count = count_temp;
-        }
+        label = static_cast<int*>(realloc(label, max_nr_class * sizeof(int)));
```

Comment:
Remove changes for this file. These changes don't affect memory leaks anyways and changes behavior in case of bad_alloc

Reply:
Actually this is the reverted code.  The current changes are for model->scaling only.  Should I revert them as well?

Reply:
No, keep the model->scaling changes. I saw these changes before you reverted them